Tests/IntegrationTests/FluentTests/TreeViewFluentTests.cs

```diff
@@ -59,10 +64,14 @@ public void TreeView_AllowReOrdering (V2TestDriver d)
                        })
                 .WaitIteration ()
                 .ScreenShot ("After re-order", _out)
-                .AssertEqual (root, tv.GetObjectOnRow (0))
-                .AssertEqual (bike, tv.GetObjectOnRow (1))
-                .AssertEqual (car, tv.GetObjectOnRow (2))
-                .AssertEqual (lorry, tv.GetObjectOnRow (3))
+                .AssertMultiple (
+                                 () =>
+                                 {
+                                     Assert.Equal (root, tv.GetObjectOnRow (0));
+                                     Assert.Equal (bike, tv.GetObjectOnRow (1));
+                                     Assert.Equal (car, tv.GetObjectOnRow (2));
+                                     Assert.Equal (lorry, tv.GetObjectOnRow (3));
+                                 })
                 .WriteOutLogs (_out);
 
         context.Stop ();
```

Comment:
There's an underlying problem here, that the old non-parallel unit tests have too:

Assert stops the test code. Thus shutdown never happens. This is the root of subsequent tests not being able to run when there's a failure. 

I wonder if there's a way to fix this via your codegen stuff such that if we do an `Assert` we ALWAYS also do an `Applicaton.ResetState(true)`?

Reply:
The codegen wraps the method in try/catch with hard stop and rethrow so should be ok